hummingbird/ml/_parse.py

```diff
@@ -283,6 +385,77 @@ def _remove_zipmap(node_list):
     return output_node_list
 
 
+def _fetch_input_slice(scope, inputs, column_indices):
+    if not isinstance(inputs, list):
+        raise TypeError("Parameter inputs must be a list.")
+    if len(inputs) == 0:
+        raise RuntimeError("Operator ArrayFeatureExtractor requires at least one inputs.")
+    if len(inputs) != 1:
+        raise RuntimeError("Operator ArrayFeatureExtractor does not support multiple input tensors.")
+
+    array_feature_extractor_operator = scope.declare_local_operator("SklearnArrayFeatureExtractor")
+    array_feature_extractor_operator.inputs = inputs
+    array_feature_extractor_operator.column_indices = column_indices
+    output_variable_name = scope.declare_local_variable("extracted_feature_columns", inputs[0].type)
+    array_feature_extractor_operator.outputs.append(output_variable_name)
+    return array_feature_extractor_operator.outputs
+
+
+def _get_column_index(i, inputs):
+    """
+    Returns a tuples (variable index, column index in that variable).
+    The function has two different behaviours, one when *i* (column index)
+    is an integer, another one when *i* is a string (column name).
+    If *i* is a string, the function looks for input name with this name and returns (index, 0).
+    If *i* is an integer, let's assume first we have two inputs
+    *I0 = FloatTensorType([None, 2])* and *I1 = FloatTensorType([None, 3])*,
+    in this case, here are the results:
+    ::
+        get_column_index(0, inputs) -> (0, 0)
+        get_column_index(1, inputs) -> (0, 1)
+        get_column_index(2, inputs) -> (1, 0)
+        get_column_index(3, inputs) -> (1, 1)
+        get_column_index(4, inputs) -> (1, 2)
+    """
+    if isinstance(i, int):
+        if i == 0:
+            # Useful shortcut, skips the case when end is None
+            # (unknown dimension)
+            return 0, 0
+        vi = 0
+        return (vi, i)
+    else:
+        raise RuntimeError("Hummingbird currently support only int columns, {} is not supported.".format(i))
+
+
+def _get_column_indices(indices, inputs):
+    """
+    Returns the requested graph inpudes based on their indices or names. See `_parse._get_column_index`.
+    Args:
+        indices: variables indices or names
+        inputs: model inputs
+
+    Returns:
+        a tuple *(variable name, list of requested indices)* if *multiple* is False, a dictionary *{ var_index: [ list of
+        requested indices ] }* if *multiple* is True
+    """
+    pt_var = None
+    pt_is = []
+    for p in indices:
+        pt_v, pt_i = _get_column_index(p, inputs)
+        pt_is.append(pt_i)
+        if pt_var is None:
+            pt_var = pt_v
+        elif pt_var != pt_v:
+            cols = [pt_var, pt_v]
```

Comment:
i  think line 450 is not necessary because we err

Reply:
Yea I thought about that too. Let me fix it.